drivers/pwm/pwm_mcux.c

```diff
@@ -135,23 +141,33 @@ static int pwm_mcux_init(const struct device *dev)
 	struct pwm_mcux_data *data = dev->data;
 	pwm_config_t pwm_config;
 	status_t status;
-	int i, err;
+	int i;
 
 	if (!device_is_ready(config->clock_dev)) {
 		LOG_ERR("clock control device not ready");
 		return -ENODEV;
 	}
 
-	err = pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
+#ifdef CONFIG_PINCTRL
+	int err = pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
 	if (err < 0) {
 		return err;
 	}
-
+#endif
 	PWM_GetDefaultConfig(&pwm_config);
 	pwm_config.prescale = config->prescale;
 	pwm_config.reloadLogic = kPWM_ReloadPwmFullCycle;
 	pwm_config.clockSource = kPWM_BusClock;
 
+	/* Since the Idle function at the system's level of */
```

Comment:
nit: block comment here should be formatted as:
```
/*
 * Some comment
 */
```
rather that multiple `/* */` styled comments

Also, would prefer some clarification in the comment here. The idle function is putting the SOC itself into a sleep state, and during the power off sequence of the LPC architecture it also puts the PWM into a sleep state, correct?